text/0171-cloudfront-redesign.md

```diff
@@ -0,0 +1,495 @@
+section provides examples of what the user experience of common (and some uncommon) use cases will be before and after the redesign.
+
+### Use Case #1 - S3 bucket origin
+
+The simplest use case is to have a single S3 bucket origin, and no customized behaviors.
+
+**Before:**
+
+```ts
+new CloudFrontWebDistribution(this, 'MyDistribution', {
+  originConfigs: [{
+    s3OriginSource: { s3BucketSource: sourceBucket },
+    behaviors : [ { isDefaultBehavior: true }],
+  }]
+});
+```
+
+**After:**
+
+```ts
+new cloudfront.Distribution(this, 'MyDistribution', {
+  origin: cloudfront.Origin.fromBucket(sourceBucket),
+});
+```
+
+### Use Case #2 - S3 bucket origin with certificate and custom behavior
+
+This example keeps the same bucket, but adds one custom behavior and a certificate.
+
+**Before:**
+
+```ts
+new CloudFrontWebDistribution(this, 'MyDistribution', {
+  originConfigs: [{
+    s3OriginSource: { s3BucketSource: sourceBucket },
+    behaviors : [
+      { isDefaultBehavior: true },
+      {
+        pathPattern: 'images/*',
+        defaultTtl: cdk.Duration.days(7),
+      }
+    ]
+  }],
+  viewerCertificate: ViewerCertificate.fromAcmCertificate(myCertificate),
+});
+```
+
+**After:**
+
+```ts
+const dist = new cloudfront.Distribution(this, 'MyDistribution', {
+  origin: cloudfront.Origin.fromBucket(sourceBucket),
+  certificate: myCertificate,
+});
+dist.origin.addBehavior('images/*', { defaultTtl: cdk.Duration.days(7) });
+```
+
+### Use Case #3 - Multi-origin, multi-behavior distribution with a Lambda@Edge function
+
+Both S3 and LoadBalancedFargateService origins, custom behaviors, and a Lambda function to top it all off.
+
+**Before:**
+
+```ts
+new CloudFrontWebDistribution(this, 'dist', {
+  originConfigs: [
+    {
+      customOriginSource: {
+        domainName: lbFargateService.loadBalancer.loadBalancerDnsName,
+        protocolPolicy: OriginProtocolPolicy.HTTPS_ONLY,
+      },
+      behaviors: [{
+        isDefaultBehavior: true,
+        allowedMethods: CloudFrontAllowedMethods.ALL,
+        forwardedValues: { queryString: true },
+        lambdaFunctionAssociations: [{
+          lambdaFunction: myFunctionVersion,
+          eventType: LambdaEdgeEventType.ORIGIN_RESPONSE,
+        }]
+      }]
+    },
+    {
+      s3OriginSource: {
+        s3BucketSource: sourceBucket,
+        originAccessIdentity: OriginAccessIdentity.fromOriginAccessIdentityName(this, 'oai', 'distOAI'),
+      },
+      behaviors: [ { pathPattern: 'static/*' } ],
+    },
+  ],
+});
+```
+
+**After:**
+
+```ts
+const dist = new cloudfront.Distribution(this, 'MyDistribution', {
+  origin: cloudfront.Origin.fromLoadBalancerV2(lbFargateService.loadBalancer, {
+    allowedMethods: AllowedMethods.ALL,
+    forwardQueryString: true,
+    edgeFunctions: [{
+      function: myFunctionVersion,
+      eventType: EventType.ORIGIN_RESPONSE,
+    }],
+  }),
+});
+dist.addOrigin(cloudfront.Origin.fromBucket(sourceBucket), { pathPattern: 'static/*' });
```

Comment:
Is the `pathPattern` required at this point?

Reply:
Yes. That example also has a mis-placed parenthesis. It should be:
```suggestion
dist.addOrigin(cloudfront.Origin.fromBucket(sourceBucket, { pathPattern: 'static/*' });
```